### PR TITLE
[FE-1604] Grids and columns #tech-debt

### PR DESCRIPTION
**Pull request by bssmain**
Migrating over the existing grid system that we have in so that we can use the same break points. This is just a straight migration with no real changes other than break points and changing to styled components.

### CODE REVIEW

File: src/Grid/grid.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+const StyledGrid = styled.div`
+  display: block;
+
+  &:last-child {
+    ${marginBottom(0)};
+  }
+
+  ${md`
+    max-width: ${1000 + GUTTER * BASE_SPACING_UNIT}px;
```
Comment (hbhrugubanda) [inline src/Grid/grid.js]: i would probably create variables for the 1000 and 1200 and define them up top

File: src/Grid/grid.story.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+  .addDecorator(withKnobs)
+  .add(
+    'Documentation',
+    withInfo({
+      text: `
+        TestInput  
+        All native input props will pass down to the input component
+      
+        ~~~js
+        <Grid />
```
Comment (hbhrugubanda) [inline src/Grid/grid.story.js]: Good to include Column as well here

Comment (hbhrugubanda) [reply src/Grid/grid.story.js]: and include more about the 12 spans making up a row

File: src/styles/spacing.js [hunk 3]
```diff
@@ -11,6 +11,14 @@ import { withPropDefault, createDefaultProp } from './utils';
  */
 export const BASE_SPACING_UNIT = 4;
 
+/**
+ * Gutter is the multiple of Base spacing unit used for the grid
+ * @constant
+ * @default 4
+ */
+
+export const GUTTER = 4;
```
Comment (hbhrugubanda) [inline src/styles/spacing.js]: `GRID_GUTTER` instead?

File: src/Grid/grid.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  ${greaterThanSm`
+    display: flex;
+    max-width: ${1200 + GUTTER * BASE_SPACING_UNIT}px;
+    ${margin(GUTTER, 'auto')};
+    align-content: center;
+    flex-wrap: wrap;
+  `};
+`;
+
+const Grid = ({ children }) => <StyledGrid>{children}</StyledGrid>;
```
Comment (ericchenak) [inline src/Grid/grid.js]: you can just use styled component in this case.
```
const Grid = styled.div``;
export default Grid;
```

File: src/Grid/column.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  )} ${greaterThanSm`
+    flex: 1 1 auto;
+    width: auto;
+    /* Styles for columns with span props */
+    ${ifProp(
+      'span',
+      css`
+        flex: 0 0 auto;
+        width: calc(
+          ${({ span }) => 100 * Math.floor(span) / 12}% -
```
Comment (ericchenak) [inline src/Grid/column.js]: you could move this part out as function
```
const transformSpan  = (span) => `calc(${100 * Math.floor(span) / 12}% - ${GUTTER * BASE_SPACING_UNIT}px);

css`
width: ${withProp('span', transformSpan)}
`

```

File: src/Grid/column.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        &:first-child,
+        &:last-child {
+          flex: 0 0 auto;
+        }
+      `,
+    )}
+  `};
+`;
+
+const Column = ({ children, span }) => (
```
Comment (ericchenak) [inline src/Grid/column.js]: not necessary to create a extra component, see comment in grid

File: src/Grid/column.js [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+import { ifProp } from 'styled-tools';
+
+import { greaterThanSm } from './../styles/breakpoints';
+import { BASE_SPACING_UNIT, GUTTER, margin } from './../styles';
+
+// constrain to 12 column spans
+const constrain = (n) => Math.max(Math.min(12, n || 1), 1);
+
+const StyledColumn = styled.div`
+  display: block;
```
Comment (ericchenak) [inline src/Grid/column.js]: could you include flex() here like spacing

File: src/Grid/column.js [hunk 8]
```diff
@@ -1,16 +1,28 @@
...
+  margin,
+  display,
+  alignItems,
+  justifyContent,
+  alignContent,
+} from './../styles';
 
 // constrain to 12 column spans
-const constrain = (n) => Math.max(Math.min(12, n || 1), 1);
+const transformSpan = (span) =>
```
Comment (ericchenak) [inline src/Grid/column.js]: you for got to constrain `span` here

File: src/Grid/column.js [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+        &:last-child {
+          flex: 0 0 auto;
+        }
+      `,
+    )}
+  `};
+`;
+
+Column.PropTypes = {
+  display: PropTypes.string,
```
Comment (ericchenak) [inline src/Grid/column.js]: add `span` to propTypes for better documentation

**Pull request merged**